dev/fattest.simplicity/src/componenttest/rules/repeater/RepeatActions.java

```diff
@@ -0,0 +1,97 @@
+     */
+    public static RepeatTests repeat(String server, TestMode otherFeatureSetsTestMode, Set<FeatureSet> allFeatureSets, FeatureSet firstFeatureSet, FeatureSet... otherFeatureSets) {
+        Set<FeatureSet> others = new HashSet<>(Arrays.asList(otherFeatureSets));
+        return repeat(server, otherFeatureSetsTestMode, allFeatureSets, firstFeatureSet, others);
+    }
+
+    /**
+     * Get a RepeatTests instance for the given FeatureSets. The first FeatureSet will be run in LITE mode. The others will be run in the mode specified by
+     * otherFeatureSetsTestMode.
+     *
+     * @param  server                   The server to repeat on
+     * @param  otherFeatureSetsTestMode The test mode to run the otherFeatureSets
+     * @param  allFeatureSets           All known FeatureSets. The features not in the current FeatureSet are removed from the repeat
+     * @param  firstFeatureSet          The first FeatureSet to repeat with. This is run in LITE mode.
+     * @param  otherFeatureSets         The other FeatureSets to repeat with. These are in the mode specified by otherFeatureSetsTestMode
+     * @return                          A RepeatTests instance
+     */
+    public static RepeatTests repeat(String server, TestMode otherFeatureSetsTestMode, Set<FeatureSet> allFeatureSets, FeatureSet firstFeatureSet,
+                                     Set<FeatureSet> otherFeatureSets) {
+        RepeatTests r = RepeatTests.with(forFeatureSet(allFeatureSets, firstFeatureSet, server, TestMode.LITE));
+        for (FeatureSet other : otherFeatureSets) {
+            r = r.andWith(forFeatureSet(allFeatureSets, other, server, otherFeatureSetsTestMode));
+        }
+        return r;
+    }
+
+    /**
+     * Get a FeatureReplacementAction instance for a given FeatureSet. It will be run in the mode specified.
+     *
+     * @param  allFeatureSets All known FeatureSets. The features not in the specified FeatureSet are removed from the repeat action
+     * @param  featureSet     The FeatureSet to repeat with.
+     * @param  server         The server to repeat on
+     * @param  testMode       The test mode to run the FeatureSet
+     * @return                A FeatureReplacementAction instance
+     */
+    public static FeatureReplacementAction forFeatureSet(Set<FeatureSet> allFeatureSets, FeatureSet featureSet, String server, TestMode testMode) {
+        FeatureReplacementAction action = null;
+        EEVersion eeVersion = featureSet.getEEVersion();
+        if (eeVersion == EEVersion.EE7)
+            action = new EE7FeatureReplacementAction();
+        else if (eeVersion == EEVersion.EE8)
+            action = new EE8FeatureReplacementAction();
+        else if (eeVersion == EEVersion.EE9)
+            action = new JakartaEE9Action();
+        else if (eeVersion == EEVersion.EE10)
+            action = new JakartaEE10Action();
+        else
+            action = new FeatureReplacementAction();
```

Comment:
If I'm reading this correctly, we define a FeatureSet for each EE version, but then actually we throw the FeatureSet away and use a specific `FeatureReplacementAction` instead?

Why don't we just use the `FeatureReplacementAction` (with all of its downsides) in the first place?

Reply:
Ok, we discussed this. We do need to call the various EE `FeatureReplacementAction` classes because they do extra stuff like transform applications for jakarta.

The featureSet does know about the features that need added and removed and those are still added to the FRA in the lines below, but it doesn't know about any transformations needed.

Some comments and more curly brackets here would be nice.